pkg/api/v1/projectref.go

```diff
@@ -0,0 +1,23 @@
+package v1
+
+import (
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1/common"
+)
+
+// ProjectDualReference encapsulates together the frequnet construct to refer to
+// a parent projet by a Kubernetes references, or an external ID, which also
+// reqwuires access credentials in such case
+type ProjectDualReference struct {
+	// "projectRef" is a reference to the parent AtlasProject resource.
+	// Mutually exclusive with the "externalProjectRef" field
+	// +kubebuilder:validation:Optional
+	Project *common.ResourceRefNamespaced `json:"projectRef,omitempty"`
```

Comment:
Non-blocking (just a nit)
```suggestion
	ProjectRef *common.ResourceRefNamespaced `json:"projectRef,omitempty"`
```
I suggest keeping the field names and the json serializer names in sync unless it is criticially necessary to deviate, otherwise this can easily cause confusion.